src/utils/resources/vm/hooks/useVMIForVM.ts

```diff
@@ -0,0 +1,29 @@
+import { VirtualMachineInstanceModelGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseVMIForVM = (
+  vmName: string,
+  vmNamespace: string,
+) => {
+  vmi: V1VirtualMachineInstance;
+  vmiLoaded: boolean;
+  vmiLoadError: any;
+};
+
+const useVMIForVM: UseVMIForVM = (vmName, vmNamespace) => {
+  const [vmi, vmiLoaded, vmiLoadError] = useK8sWatchResource<V1VirtualMachineInstance>({
+    groupVersionKind: VirtualMachineInstanceModelGroupVersionKind,
+    isList: false,
+    name: vmName,
+    namespace: vmNamespace,
+  });
+
+  return {
+    vmi: vmName === vmi?.metadata?.name && vmNamespace === vmi?.metadata?.namespace && vmi,
+    vmiLoaded: vmiLoaded,
```

Comment:
can be simply

{vmi, vmiLoaded, vmiLoadError} no need for {[name]: [name]}

Reply:
Fixed!